src/internal.c

```diff
@@ -201,13 +201,23 @@ static void sha256_midstate(struct sha256_ctx *ctx, struct sha256 *res)
     size_t i;
 
 #ifdef CCAN_CRYPTO_SHA256_USE_MBEDTLS
+#ifndef CONFIG_MBEDTLS_HARDWARE_SHA
+#define SHA_CTX_STATE c.MBEDTLS_PRIVATE(state)
+#else
 #define SHA_CTX_STATE c.state
+#endif
 #else
 #define SHA_CTX_STATE s
 #endif
 
+#if defined(CCAN_CRYPTO_SHA256_USE_MBEDTLS) && \
+    defined(MBEDTLS_SHA256_ALT) && !defined(SOC_SHA_SUPPORT_PARALLEL_ENG)
+    /* HW: Already big endian */
```

Comment:
I think we should make this:

```
#if defined(CCAN_CRYPTO_SHA256_USE_MBEDTLS) && \
    ((defined(MBEDTLS_SHA256_ALT) && !defined(SOC_SHA_SUPPORT_PARALLEL_ENG)) || defined(SOC_SHA_ENDIANNESS_BE)
```

Which will kick in when the idf sdk is updated in the future, what do you think?

Reply:
I've updated it to be:

```
#if defined(CCAN_CRYPTO_SHA256_USE_MBEDTLS) && \                                
    defined(MBEDTLS_SHA256_ALT) && (!defined(SOC_SHA_SUPPORT_PARALLEL_ENG) || !defined(SOC_SHA_ENDIANNESS_BE))                                                                                                     
    /* HW: Already big endian */      
```

also updated the other for symmetry:
```
#if defined(CCAN_CRYPTO_SHA256_USE_MBEDTLS) && \                                
    defined(MBEDTLS_SHA256_ALT) && (defined(SOC_SHA_SUPPORT_PARALLEL_ENG) || defined(SOC_SHA_ENDIANNESS_BE))                                                                                                       
    /* HW sha engine doesn't allow to extract the midstate */                   
    ctx.c.mode = ESP_MBEDTLS_SHA256_SOFTWARE;      
```